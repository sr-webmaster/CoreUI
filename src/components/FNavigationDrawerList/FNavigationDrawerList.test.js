import { mount } from '@vue/test-utils'
import Component from '@freshinup/core-ui/src/components/FNavigationDrawerList'

const items = [
  { title: 'Users', to: { path: '/admin/users' }, action: 'far fa-user' },
  { title: 'Teams', to: { path: '/admin/teams' } },
  { title: 'Companies', to: { path: '/admin/companies' } }
]

describe('FNavigationDrawerList', () => {
  it('snapshot default', async () => {
    const wrapper = mount(Component, {
      propsData: {
        items
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('.v-list .v-list__tile__action').exists()).toBe(true)
  })
  it('snapshot no actions', async () => {
    const wrapper = mount(Component, {
      propsData: {
        items,
        noActions: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('.v-list .v-list__tile__action').exists()).toBe(false)
  })
})

describe('ColoredFNavigationDrawerList', () => {
  it('snapshot default', async () => {
    const wrapper = mount(Component, {
      propsData: {
        items,
        backgroundActiveColor: 'red',
        foregroundActiveColor: 'white'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('.v-list .v-list__tile__action').exists()).toBe(true)
  })

  it('snapshot no actions', async () => {
    const wrapper = mount(Component, {
      propsData: {
        items,
        noActions: true,
        backgroundActiveColor: 'red',
        foregroundActiveColor: 'white'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('.v-list .v-list__tile__action').exists()).toBe(false)
  })

  it('has colored active menu item', async () => {
    const wrapper = mount(Component, {
      propsData: {
        items,
        backgroundActiveColor: 'red',
        foregroundActiveColor: 'white'
      }
    })
    expect(wrapper.props().backgroundActiveColor).toBe('red')
    expect(wrapper.props().foregroundActiveColor).toBe('white')
  })
})
