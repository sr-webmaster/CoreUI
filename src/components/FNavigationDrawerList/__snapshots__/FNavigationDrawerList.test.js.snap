// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColoredFNavigationDrawerList snapshot default 1`] = `
<div
  class="v-list theme--light"
  role="list"
>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <div
        class="v-list__tile__action"
      >
        <i
          aria-hidden="true"
          class="v-icon far fa-user theme--light"
        />
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Users
        </div>
      </div>
    </router-link>
  </div>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <div
        class="v-list__tile__action"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light"
        />
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Teams
        </div>
      </div>
    </router-link>
  </div>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <div
        class="v-list__tile__action"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light"
        />
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Companies
        </div>
      </div>
    </router-link>
  </div>
</div>
`;

exports[`ColoredFNavigationDrawerList snapshot no actions 1`] = `
<div
  class="v-list theme--light"
  role="list"
>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <!---->
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Users
        </div>
      </div>
    </router-link>
  </div>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <!---->
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Teams
        </div>
      </div>
    </router-link>
  </div>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <!---->
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Companies
        </div>
      </div>
    </router-link>
  </div>
</div>
`;

exports[`FNavigationDrawerList snapshot default 1`] = `
<div
  class="v-list theme--light"
  role="list"
>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <div
        class="v-list__tile__action"
      >
        <i
          aria-hidden="true"
          class="v-icon far fa-user theme--light"
        />
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Users
        </div>
      </div>
    </router-link>
  </div>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <div
        class="v-list__tile__action"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light"
        />
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Teams
        </div>
      </div>
    </router-link>
  </div>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <div
        class="v-list__tile__action"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light"
        />
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Companies
        </div>
      </div>
    </router-link>
  </div>
</div>
`;

exports[`FNavigationDrawerList snapshot no actions 1`] = `
<div
  class="v-list theme--light"
  role="list"
>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <!---->
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Users
        </div>
      </div>
    </router-link>
  </div>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <!---->
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Teams
        </div>
      </div>
    </router-link>
  </div>
  <div
    role="listitem"
  >
    <router-link
      class="v-list__tile v-list__tile--link theme--light"
    >
      <!---->
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          Companies
        </div>
      </div>
    </router-link>
  </div>
</div>
`;
